### DISCUSSION TITLE
Class component to Hooks

### DISCUSSION THREAD

**Discussion by vctormb** [ritikbanger/react18-input-otp#29] [Ideas]
Hello!

First I'd like to thank you for this awesome library!

I noticed that the library uses React Class components. Will this be refactored to use React Hooks instead or there's no plannings for that?

Thanks!

**Comment by ritikbanger** [comment]
Hi @vctormb, Thanks for using the library. As of now, I am not planning to move to function right now as more than 3lac devs are using it and it might break something but surely it will make the code a lot better. Would you like to work with me on this to move to hooks?

**Reply by vctormb** [reply]
I'm wondering if we should add some tests with react-testing-library and jest before doing any refactoring 🤔 What do you think?

**Reply by ritikbanger** [reply]
Why not!

**Comment by ritikbanger** [comment]
Could you start with some test on a separate branch? And i would commit on the same?
